Enable RuboCop style cops

rafaelfranca opened the pull request:
- enable [Style/AmbiguousEndlessMethodDefinition](https://docs.rubocop.org/rubocop/cops_style.html#styleambiguousendlessmethoddefinition)
- enable [Style/AndOr](https://docs.rubocop.org/rubocop/cops_style.html#styleandor)
- enable [Style/ArrayCoercion](https://docs.rubocop.org/rubocop/cops_style.html#stylearraycoercion)
- enable [Style/Attr](https://docs.rubocop.org/rubocop/cops_style.html#styleattr)
- enable [Style/BlockComments](https://docs.rubocop.org/rubocop/cops_style.html#styleblockcomments)
- enable [Style/CharacterLiteral](https://docs.rubocop.org/rubocop/cops_style.html#stylecharacterliteral)
- enable [Style/ClassAndModuleChildren](https://docs.rubocop.org/rubocop/cops_style.html#styleclassandmodulechildren)
- enable [Style/ClassMethodsDefinitions](https://docs.rubocop.org/rubocop/cops_style.html#styleclassmethodsdefinitions)
- enable [Style/ClassMethods](https://docs.rubocop.org/rubocop/cops_style.html#styleclassmethods)
- enable [Style/ClassVars](https://docs.rubocop.org/rubocop/cops_style.html#styleclassvars)
- enable [Style/ColonMethodCall](https://docs.rubocop.org/rubocop/cops_style.html#stylecolonmethodcall)
- enable [Style/ColonMethodDefinition](https://docs.rubocop.org/rubocop/cops_style.html#stylecolonmethoddefinition)
- enable [Style/CombinableDefined](https://docs.rubocop.org/rubocop/cops_style.html#stylecombinabledefined)
- enable [Style/DateTime](https://docs.rubocop.org/rubocop/cops_style.html#styledatetime)
- enable [Style/DefWithParentheses](https://docs.rubocop.org/rubocop/cops_style.html#styledefwithparentheses)
- enable [Style/DigChain](https://docs.rubocop.org/rubocop/cops_style.html#styledigchain)
- enable [Style/EndlessMethod](https://docs.rubocop.org/rubocop/cops_style.html#styleendlessmethod)
- enable [Style/GlobalVars](https://docs.rubocop.org/rubocop/cops_style.html#styleglobalvars)
- enable [Style/HashSyntax](https://docs.rubocop.org/rubocop/cops_style.html#stylehashsyntax)
- enable [Style/MagicCommentFormat](https://docs.rubocop.org/rubocop/cops_style.html#stylemagiccommentformat)
- enable [Style/MethodDefParentheses](https://docs.rubocop.org/rubocop/cops_style.html#stylemethoddefparentheses)
- enable [Style/RedundantBegin](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantbegin)
- enable [Style/RedundantReturn](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantreturn)
- enable [Style/Semicolon](https://docs.rubocop.org/rubocop/cops_style.html#stylesemicolon)
- enable [Style/SymbolLiteral](https://docs.rubocop.org/rubocop/cops_style.html#stylesymbolliteral)
- enable [Style/TrailingBodyOnClass](https://docs.rubocop.org/rubocop/cops_style.html#styletrailingbodyonclass)
- enable [Style/TrailingBodyOnMethodDefinition](https://docs.rubocop.org/rubocop/cops_style.html#styletrailingbodyonmethoddefinition)
- enable [Style/TrailingBodyOnModule](https://docs.rubocop.org/rubocop/cops_style.html#styletrailingbodyonmodule)
- enable [Style/TrailingMethodEndStatement](https://docs.rubocop.org/rubocop/cops_style.html#styletrailingmethodendstatement)
- enable [Style/UnlessElse](https://docs.rubocop.org/rubocop/cops_style.html#styleunlesselse)
- enable [Style/VariableInterpolation](https://docs.rubocop.org/rubocop/cops_style.html#stylevariableinterpolation)

rafaelfranca committed:
Enable RuboCop style cops

- enable [Style/AmbiguousEndlessMethodDefinition](https://docs.rubocop.org/rubocop/cops_style.html#styleambiguousendlessmethoddefinition)
- enable [Style/AndOr](https://docs.rubocop.org/rubocop/cops_style.html#styleandor)
- enable [Style/ArrayCoercion](https://docs.rubocop.org/rubocop/cops_style.html#stylearraycoercion)
- enable [Style/Attr](https://docs.rubocop.org/rubocop/cops_style.html#styleattr)
- enable [Style/BisectedAttrAccessor](https://docs.rubocop.org/rubocop/cops_style.html#stylebisectedattraccessor)
- enable [Style/BlockComments](https://docs.rubocop.org/rubocop/cops_style.html#styleblockcomments)
- enable [Style/CharacterLiteral](https://docs.rubocop.org/rubocop/cops_style.html#stylecharacterliteral)
- enable [Style/ClassAndModuleChildren](https://docs.rubocop.org/rubocop/cops_style.html#styleclassandmodulechildren)
- enable [Style/ClassMethodsDefinitions](https://docs.rubocop.org/rubocop/cops_style.html#styleclassmethodsdefinitions)
- enable [Style/ClassMethods](https://docs.rubocop.org/rubocop/cops_style.html#styleclassmethods)
- enable [Style/ClassVars](https://docs.rubocop.org/rubocop/cops_style.html#styleclassvars)
- enable [Style/ColonMethodCall](https://docs.rubocop.org/rubocop/cops_style.html#stylecolonmethodcall)
- enable [Style/ColonMethodDefinition](https://docs.rubocop.org/rubocop/cops_style.html#stylecolonmethoddefinition)
- enable [Style/CombinableDefined](https://docs.rubocop.org/rubocop/cops_style.html#stylecombinabledefined)
- enable [Style/DateTime](https://docs.rubocop.org/rubocop/cops_style.html#styledatetime)
- enable [Style/DefWithParentheses](https://docs.rubocop.org/rubocop/cops_style.html#styledefwithparentheses)
- enable [Style/DigChain](https://docs.rubocop.org/rubocop/cops_style.html#styledigchain)
- enable [Style/EndlessMethod](https://docs.rubocop.org/rubocop/cops_style.html#styleendlessmethod)
- enable [Style/GlobalVars](https://docs.rubocop.org/rubocop/cops_style.html#styleglobalvars)
- enable [Style/HashSyntax](https://docs.rubocop.org/rubocop/cops_style.html#stylehashsyntax)
- enable [Style/MagicCommentFormat](https://docs.rubocop.org/rubocop/cops_style.html#stylemagiccommentformat)
- enable [Style/MethodDefParentheses](https://docs.rubocop.org/rubocop/cops_style.html#stylemethoddefparentheses)
- enable [Style/RedundantBegin](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantbegin)
- enable [Style/RedundantReturn](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantreturn)
- enable [Style/RedundantSelf](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantself)
- enable [Style/Semicolon](https://docs.rubocop.org/rubocop/cops_style.html#stylesemicolon)
- enable [Style/SymbolLiteral](https://docs.rubocop.org/rubocop/cops_style.html#stylesymbolliteral)
- enable [Style/TrailingBodyOnClass](https://docs.rubocop.org/rubocop/cops_style.html#styletrailingbodyonclass)
- enable [Style/TrailingBodyOnMethodDefinition](https://docs.rubocop.org/rubocop/cops_style.html#styletrailingbodyonmethoddefinition)
- enable [Style/TrailingBodyOnModule](https://docs.rubocop.org/rubocop/cops_style.html#styletrailingbodyonmodule)
- enable [Style/TrailingMethodEndStatement](https://docs.rubocop.org/rubocop/cops_style.html#styletrailingmethodendstatement)
- enable [Style/UnlessElse](https://docs.rubocop.org/rubocop/cops_style.html#styleunlesselse)
- enable [Style/VariableInterpolation](https://docs.rubocop.org/rubocop/cops_style.html#stylevariableinterpolation)

burke commented on test/fixtures/full_config.yml:
```diff
@@ -2731,7 +2731,7 @@ Style/ClassMethodsDefinitions:
   Description: Avoid the use of class variables.
   StyleGuide: "#no-class-vars"
-  Enabled: false
+  Enabled: true
```
I have also always kinda disliked this rule although I will admit it is not common for them to be the right call

rafaelfranca replied:
I'll keep this in given the thread-safety and easy to fix alternative.

burke commented on test/fixtures/full_config.yml:
```diff
@@ -4379,7 +4379,7 @@ Style/StringHashKeys:
   Description: Checks if uses of quotes match the configured preference.
   StyleGuide: "#consistent-string-literals"
-  Enabled: false
+  Enabled: true
```
I have always despised this rule but I'm not gonna spend any more energy fighting it

rafaelfranca replied:
I'll keep this out

hmcguire-shopify commented on test/fixtures/full_config.yml:
```diff
@@ -2608,14 +2608,14 @@ Style/BitwisePredicate:
     ugly). Prefer {...} over do...end for single-line blocks.
   StyleGuide: "#single-line-blocks"
-  Enabled: false
+  Enabled: true
```
I find requiring `do end` for multiline blocks makes chaining look worse:

```
map do |x|
end.filter do |x|
end
```

vs

```
map { |x|
}.filter { |x|
}
```

It looks like there's a `braces_for_chaining` style, but I don't know if that's strictly better

rafaelfranca replied:
Good point. I'll keep this out.

rafaelfranca commented on test/fixtures/full_config.yml:
indeed. I'd put that back

rafaelfranca committed:
Disable Style/BisectedAttrAccessor

It conflicts with Sorbet signatures.

rafaelfranca committed:
Disable Style/RedundantSelf

It conflicts with RBS inline comments.

For example:

```
self #: as untyped
  .some_method
```

rafaelfranca committed:
Disable Style/CollectionQuerying

It isn't safe to autocorrect.

rafaelfranca committed:
Disable Style/ClassAndModuleChildren

We don't want to enforce this style for now.

We have too many constants that have very deep nesting and it
makes hard to read the code.

So we shouuld allow the compact style as well.

rafaelfranca committed:
Disable Style/DateTime

It isn't safe to autocorrect.
